BraintreeCore/src/main/java/com/braintreepayments/api/Configuration.kt

```diff
@@ -0,0 +1,410 @@
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val googlePayAuthorizationFingerprint = googlePayConfiguration.googleAuthorizationFingerprint
+
+    /**
+     * @return the current Google Pay environment.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val googlePayEnvironment = googlePayConfiguration.environment
+
+    /**
+     * @return the Google Pay display name to show to the user.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val googlePayDisplayName = googlePayConfiguration.displayName
+
+    /**
+     * @return a list of supported card networks for Google Pay.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val googlePaySupportedNetworks = googlePayConfiguration.supportedNetworks
+
+    /**
+     * @return the PayPal Client ID used by Google Pay.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val googlePayPayPalClientId = googlePayConfiguration.paypalClientId
+
+    /**
+     * @return [String] url of the Braintree analytics service.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val analyticsUrl = analyticsConfiguration.url
+
+    /**
+     * @return `true` if analytics are enabled, `false` otherwise.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val isAnalyticsEnabled = analyticsConfiguration.isEnabled
+
+    /**
+     * @return `true` if Visa Checkout is enabled for the merchant account; `false` otherwise.
+     */
+    open val isVisaCheckoutEnabled = visaCheckoutConfiguration.isEnabled
+
+    /**
+     * @return the Visa Checkout supported networks enabled for the merchant account.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val visaCheckoutSupportedNetworks = visaCheckoutConfiguration.acceptedCardBrands
+
+    /**
+     * @return the Visa Checkout API key configured in the Braintree Control Panel.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val visaCheckoutApiKey = visaCheckoutConfiguration.apiKey
+
+    /**
+     * @return the Visa Checkout External Client ID configured in the Braintree Control Panel.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val visaCheckoutExternalClientId = visaCheckoutConfiguration.externalClientId
+
+    /**
+     * Check if a specific feature is enabled in the GraphQL API.
+     *
+     * @param feature The feature to check.
+     * @return `true` if GraphQL is enabled and the feature is enabled, `false` otherwise.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    fun isGraphQLFeatureEnabled(feature: String) = graphQLConfiguration.isFeatureEnabled(feature)
+
+    /**
+     * @return the GraphQL url.
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    val graphQLUrl = graphQLConfiguration.url
+
+    /**
+     * @return `true` if Samsung Pay is enabled; `false` otherwise.
+     */
+    open val isSamsungPayEnabled = samsungPayConfiguration.isEnabled
```

Comment:
On iOS, we've gotten into the practice of organizing Swift files by scope, so basically grouping methods and properties by access level. For example,[ these types of MARKs](https://github.com/braintree/braintree_ios/blob/c449e0b81c78fcae47bb02cfb8f987aed2a19996/Sources/BraintreeCore/BTAPIClient.swift#L20). We outline it in our [style guide too](https://hackmd.io/kIyOnIFIRrKY4b5wQ0dUpw?both#Code-Organization).

This grouping will probably help readability on Android too.

Reply:
Yeah we should definitely align our style guide with y'alls at some point. We may even end up migrating feature related configuration code into extensions like in iOS since Kotlin has a similar feature.

Reply:
Added a ticket for this: No. 2211